bakerydemo/blog/models.py

```diff
@@ -138,10 +138,29 @@ def tag_archive(self, request, tag=None):
                 messages.add_message(request, messages.INFO, msg)
             return redirect(self.url)
 
-        blogs = BlogPage.objects.filter(tags=tag).live().descendant_of(self)
+        blogs = self.blogs(tag=tag)
 
         context = {
             'tag': tag,
             'blogs': blogs
         }
         return render(request, 'blog/blog_index_page.html', context)
+
+    def blogs(self, tag=None):
+        """
+        Return the child BlogPage objects for this BlogPageIndex. Optional
+        filter by tag
+        """
+        blogs = BlogPage.objects.live().descendant_of(self)
+        if tag:
+            blogs = blogs.filter(tags=tag)
+        return blogs
+
+    def get_child_tags(self):
+        """
+        Returns the list of Tags for child pages.
+        """
+        tags = [x.get_tags for x in self.blogs()]
```

Comment:
This is a nice feature! But I think this might be one of those areas where dense list comprehensions save a bit of code but make the code less easy to understand. I might use for loops here, or a for loop combined with list comprehension. I'll merge this and take a crack at it.